### ISSUE TITLE
Cannot publish using a generated sas token

### ISSUE THREAD

**Issue by ErikMogensen** [Azure/api-management-developer-portal#1196]
## Bug description

When creating a token using the following code from the scripts.v2/migrate.js file it works when calling the direct REST API, but the token does not work for publishing the portal.

```
 async function generateSASToken(id, key, expiresIn = 3600) {
    const now = moment.utc(moment());
    const expiry = now.clone().add(expiresIn, 'seconds');
    const expiryString = expiry.format(`YYYY-MM-DD[T]HH:mm:ss.SSSSSSS[Z]`);

    const dataToSign = `${id}\n${expiryString}`;
    const signedData = crypto.createHmac('sha512', key).update(dataToSign).digest('base64');
    return `SharedAccessSignature uid=${id}&ex=${expiryString}&sn=${signedData}`;
}
```

However the publishing works if an access token is used which is generated using the Management API blade in the Azure Portal.

## Reproduction steps

1. Generate a token using the code above. 
2. Call the publish using the generated token :
 POST https://<service>.developer.azure-api.net/publish HTTP/1.1
content-type: application/json 
Authorization: SharedAccessSignature i45983457.........
4. 200 is received
5. Wait three minutes to make sure that the portal was not published.

## Expected behavior

Expected the portal to be published.

## Is your portal managed or self-hosted?

Managed

## API Management service name

I'll send you the name.

## Environment

 - Operating system: Windows 10
 - Node : v14.16.0

**Comment by azaslonov** [comment]
Hi Erik, yes, there is a regression. We're working on it.

**Comment by azaslonov** [comment]
Most of public services have been upgraded, the issue should be gone.

**Comment by ErikMogensen** [comment]
I just tried this on an instance in West Europe with:

  "CodeVersion": "20210402201829",
  "Version": "0.19.1172.0"

But it is still not working :(

Is this a backend update? Which version number is it included in?

**Comment by ErikMogensen** [comment]
If you fix #1039, "Publish using ARM REST APIs", I won't care about this issue.

**Comment by ErikMogensen** [comment]
SInce #1039 works you may close this as far as I am concerned.
